app/pages/game/[slug].vue

```diff
@@ -1,30 +1,55 @@
 <script setup lang="ts">
-import { useRoute, createError } from '#imports'
+import { useRoute } from '#imports'
 import Loader from 'blocks/Loader/Loader.vue'
-import { useGetLeaderboardBySlug } from '~/composables/api'
-import LeaderboardInfo from '~/components/blocks/LeaderboardInfo/LeaderboardInfo.vue'
+import CategorySelect from '~/components/blocks/RunsPage/CategorySelect/CategorySelect.vue'
+import RunsHeader from '~/components/blocks/RunsPage/Header/Header.vue'
+import RunsTable from '~/components/blocks/RunsPage/RunsTable/RunsTable.vue'
+import {
+  useGetCategoryBySlug,
+  useGetLeaderboardBySlug,
+} from '~/composables/api'
+import type { CategoryViewModel } from '~~/lib/api/data-contracts'
 const {
   params: { slug },
+  query: { category: categorySlug },
 } = useRoute()
 
+if (typeof slug !== 'string' || typeof categorySlug !== 'string') {
```

Comment:
I think instead of doing this you'll want to just cast to string and check the truthiness, and then error if it's falsey.